### PR TITLE
feat(prisma-fmt): implement native types in text_document_completion

### PR DESCRIPTION
**Pull request by tomhoule**
Originally to fix https://github.com/prisma/language-tools/issues/1195,
but that would also let us remove the native_types method on prisma-fmt.

### CODE REVIEW

File: prisma-fmt/tests/text_document_completion/tests.rs [hunk 1]
```diff
@@ -17,18 +17,20 @@ scenarios! {
...
-    extended_indexes_operators_cockroach_gin
+    extended_indexes_types_cockroach
+    extended_indexes_types_mongo
+    extended_indexes_types_mysql
+    extended_indexes_types_postgres
+    extended_indexes_types_sqlite
+    extended_indexes_types_sqlserver
     language_tools_relation_directive
+    native_type_empty
+    native_type_partially_typed
```
Comment (tomhoule) [inline prisma-fmt/tests/text_document_completion/tests.rs:33]: These are the two new lines. The rest is just sorted.

File: prisma-fmt/src/text_document_completion.rs [hunk 2]
```diff
@@ -81,6 +97,41 @@ fn push_ast_completions(
...
+                    ""
+                } else {
+                    "($0)"
+                };
+
+                completion_list.items.push(CompletionItem {
+                    label: name.to_owned(),
+                    kind: Some(CompletionItemKind::CONSTRUCTOR),
+                    insert_text: Some(format!("{name}{args}")),
+                    insert_text_format: Some(InsertTextFormat::SNIPPET),
```
Comment (Jolg42) [inline prisma-fmt/src/text_document_completion.rs:130]: Probably doesn't need to be a snippet here, plain text should be enough? (no need to put cursor inside `()`)

Though technically, we could do it for `@db.VarChar()` but then the completion needs to include `$0` like `@db.VarChar($0)`

**Review by Jolg42 — COMMENTED** [review]
This is a completion already done on TS side.
I think we should sync when something like this is implemented in prisma-fmt because we could remove the TS code at least.

File: prisma-fmt/src/text_document_completion.rs (hunk 2 above)
Comment (tomhoule) [reply prisma-fmt/src/text_document_completion.rs:130]: >Though technically, we could do it for @db.VarChar() but then the completion needs to include $0 like @db.VarChar($0)

This is what I tried to do (see line 123), but I could be wrong. This needs to be tested end-to-end by getting it into an editor before we merge.

**Review by tomhoule — COMMENTED** [review]
@Jolg42 yep, we have to test this beforehand, and make the changes in the language server, but I think that would be good, it shrinks the API of prisma-fmt-wasm and makes things more uniform.

File: prisma-fmt/src/text_document_completion.rs (hunk 2 above)
Comment (Jolg42) [reply prisma-fmt/src/text_document_completion.rs:130]: I missed that 😅 That should work then 👍🏼 
But we should have a setup where we could easily test this end to end at least in dev.

Comment (tomhoule) [reply prisma-fmt/src/text_document_completion.rs:130]: Yes, I think Julius worked on that a bit. I'll try to see how easy I can make it as soon as I have some time.

**Comment by tomhoule** [comment]
Let's drop this if we aren't working on it.

**Pull request closed without merging**